src/Pdf/Layouts/TextLayout.cs

```diff
@@ -83,46 +128,26 @@ private static List<string> WrapText(string text, TextStyle style, float maxWidt
 
     public LayoutResult Layout(LayoutContext context)
     {
-        if (_drawn || _lines.Count == 0)
-        {
-            return new LayoutResult([], LayoutStatus.IsFullyDrawn);
-        }
-
         if (_wrappedLines.Count == 0)
         {
             _wrappedLines = _lines.SelectMany(line => WrapText(line, Style, context.Available.Width)).ToList();
         }
 
         var listDrawables = new List<IDrawable>();
-        var halfLineHeight = (Style.LineHeight * Style.FontSize - Style.FontSize) / 2;
         var top = context.Available.Top;
 
         while (_currentIndex < _wrappedLines.Count)
         {
-            var textLineLocation = top + halfLineHeight - Style.ToFont().Metrics.Ascent;
-
-            if (_currentIndex == 0)
-            {
-                textLineLocation += Style.ParagraphSpacingBefore;
-            }
-
-            var textRect = new SKRect(context.Available.Left, textLineLocation, context.Available.Right,
-                textLineLocation);
-            var allocationRectBottom = textLineLocation + halfLineHeight + Style.ToFont().Metrics.Descent;
-
-            if (_currentIndex == _wrappedLines.Count - 1)
-            {
-                allocationRectBottom += Style.ParagraphSpacingAfter;
-            }
+            var textRect = MeasureTextLineRect(context.Available, top, _currentIndex);
```

Comment:
with all the text layout changes I also want to double-check the test files to ensure line height, font sizes, and paragraph spacings are respected